projects/js-packages/publicize-components/src/components/social-post-modal/modal.tsx

```diff
@@ -1,57 +1,75 @@
 import { useAnalytics } from '@automattic/jetpack-shared-extension-utils';
-import { Modal, PanelRow, Button } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { Button, Modal, PanelRow } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
-import { useCallback, useReducer } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
 import { __, _x } from '@wordpress/i18n';
 import { close } from '@wordpress/icons';
+import { store as socialStore } from '../../social-store';
 import { PreviewSection } from './preview-section';
 import { SettingsSection } from './settings-section';
 import styles from './styles.module.scss';
 
 /**
  * The Social Post Modal component.
  *
- * @return {import('react').ReactNode} - Social Post Modal component.
+ * @param {{title:string}} props - The component props.
+ *
+ * @return - Social Post Modal component.
+ */
+function RenderSocialPostModal( { title }: { title: string } ) {
+	const { closeSharePostModal } = useDispatch( socialStore );
```

Comment:
If this is going to be another component, shall we pass this in? But equally we could leave this code where it is for now.

Reply:
I have moved the logic to props.